src/Nethermind/Nethermind.BeaconNode.Peering/MothraPeeringWorker.cs

```diff
@@ -244,27 +315,52 @@ private void OnGossipReceived(ReadOnlySpan<byte> topicUtf8, ReadOnlySpan<byte> d
 
         private void OnPeerDiscovered(ReadOnlySpan<byte> peerUtf8)
         {
+            string peerId = Encoding.UTF8.GetString(peerUtf8);
             try
             {
-                if (_logger.IsInfo()) Log.PeerDiscovered(_logger, Encoding.UTF8.GetString(peerUtf8), null);
+                if (_logger.IsInfo()) Log.PeerDiscovered(_logger, peerId, null);
+                if (!ThreadPool.QueueUserWorkItem(HandlePeerDiscoveredAsync, peerId, true))
```

Comment:
will that start throwing a lot of exceptions when we discover 1000 nodes per second?
In Eth1 we just add them to a list and there is a separate loop responsible for establishing connections

Reply:
Yes, a blocking queue with a separate loop for processing is a better approach. I think I have a comment somewhere about that. I will add as a separate work item to the project so we know it needs to be done.

Reply:
Oh, I think I understand your comment now -- you might discover / learn about thousands of nodes, but only connect to a relatively small number of them. Yes, we probably need to do that, although for the small test net, I'm just connecting as is.

But, I did end up changing the processing to use a queue, the new core 3 Channel<T> class, which seems to be the best producer/consumer at the moment. Rather than a separate work item, I just did it; much nicer than spinning up threadpool.

Decoupling it this way will also make it easier to switch to a discovery pool as well, with threads maintaining a target number of open connections.